environment/src/main/java/com/sequenceiq/environment/environment/service/EnvironmentLoadBalancerService.java

```diff
@@ -67,6 +86,32 @@ public void updateLoadBalancerInEnvironmentAndStacks(EnvironmentDto environmentD
         reactorFlowManager
             .triggerLoadBalancerUpdateFlow(environmentDto, environment.getId(), environment.getName(), environment.getResourceCrn(),
                 environmentLbDto.getEndpointAccessGateway(), environmentLbDto.getEndpointGatewaySubnetIds(), userCrn);
+
+        FlowLogResponse flowLogResponse = getLatestLoadBalancerUpdateFlowConfigLogs(environment.getResourceCrn());
+        if (flowLogResponse == null || hasFlowFailed(List.of(flowLogResponse))) {
+            String errorMsg;
+            if (flowLogResponse == null) {
+                errorMsg = String.format("Unable to find %s flow for tracking", LoadBalancerUpdateFlowConfig.class.getSimpleName());
+            } else {
+                response.setFlowId(flowLogResponse.getFlowId());
+                errorMsg = String.format("Flow in %s state. Please review environment logs.", flowLogResponse.getCurrentState());
+            }
+            LOGGER.error(errorMsg);
+            response.setError(errorMsg);
+            response.setStatus(LoadBalancerUpdateStatus.FAILED);
+        } else {
+            response.setFlowId(flowLogResponse.getFlowId());
+            response.setStatus(LoadBalancerUpdateStatus.IN_PROGRESS);
+        }
+
+        return response;
+    }
+
+    @VisibleForTesting
+    FlowLogResponse getLatestLoadBalancerUpdateFlowConfigLogs(String environmentCrn) {
+        return flowService.getFlowLogsByCrnAndType(environmentCrn, LoadBalancerUpdateFlowConfig.class).stream()
+            .max(Comparator.comparing(FlowLogResponse::getCreated))
+            .orElse(null);
```

Comment:
you should just return with optional, that's the purpose of it

Reply:
Very true. I'll fix that.